R/plot_with_settings.R

```diff
@@ -378,14 +373,30 @@ plot_with_settings_srv <- function(id,
     p_height <- reactive(`if`(!is.null(input$height), input$height, height[1]))
     p_width <- reactive(`if`(!is.null(input$width), input$width, default_slider_width()[1]))
     output$plot_main <- renderPlot(
-      plot_reactive(),
+      {
+        if (plot_type() == "grob") {
+          grid::grid.newpage()
+          grid::grid.draw(plot_reactive())
+        } else {
+          plot_reactive()
+        }
```

Comment:
repeated code - please create an utility function

```suggestion
        if (plot_type() == "grob") {
          grid::grid.newpage()
          grid::grid.draw(plot_reactive())
        } else {
          plot_reactive()
        }
```